dvc/repo/__init__.py

```diff
@@ -293,7 +294,17 @@ def graph(self, stages=None, from_directory=None):
 
         for stage in stages:
             for out in stage.outs:
-                existing = [o.stage for o in outs if o.path == out.path]
+
+                existing = []
+                for o in outs:
+                    if o.path == out.path:
+                        existing.append(o.stage)
+
+                    if (
+                        o.path + os.sep in out.path
```

Comment:
you are probably looking for startswith. Using `in` is error prone. For example `o.path == 'c'` and `out.path == '/a/b/c'`.

Reply:
Done